pkg/providers/vsphere/vmprovider_vm.go

```diff
@@ -1102,6 +1111,47 @@ func (vs *vSphereVMProvider) reconcilePowerState(
 	return nil
 }
 
+func (vs *vSphereVMProvider) reconcileSnapshot(
+	vmCtx pkgctx.VirtualMachineContext,
+	vcVM *object.VirtualMachine) error {
+	// Skip creating a new snapshot if the VM already points to the desired snapshot.
+	if vmCtx.VM.Spec.CurrentSnapshot == nil || reflect.DeepEqual(vmCtx.VM.Spec.CurrentSnapshot, vmCtx.VM.Status.CurrentSnapshot) {
+		return nil
+	}
+
+	// get VirtualMachineSnapshot object
+	vmSnapshot, err := getVirtualMachineSnapShotObject(vmCtx, vs.k8sClient)
+	if err != nil {
+		return err
+	}
+
+	snapArgs := virtualmachine.SnapshotArgs{
+		VMCtx:      vmCtx,
+		VcVM:       vcVM,
+		VMSnapshot: vmSnapshot,
+	}
+
+	vmCtx.Logger.Info("Taking a snapshot of VM Service managed VM")
```

Comment:
vvNit: ALL VMs that we are reconciling are VM Service managed VMs.  Just say `Creating a new snapshot of the virtual machine` or something like that.

Reply:
ack